data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.IndexParametersConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.model.ServiceInfo;
+import org.opensearch.dataprepper.plugins.source.opensearch.service.ElasticSearchService;
+import org.opensearch.dataprepper.plugins.source.opensearch.service.HostsService;
+import org.opensearch.dataprepper.plugins.source.opensearch.service.OpenSearchService;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.OpenSearchTimerWorker;
+import java.util.Timer;
+
+/**
+ *  Service that will call both worker classes
+ */
+public class OpenSearchSourceService {
+
+    private final OpenSearchService openSearchService;
+
+    private final ElasticSearchService elasticSearchService;
+
+    private final HostsService hostsService;
+
+    private final OpenSearchSourceConfiguration sourceConfig;
+
+    private final BufferAccumulator bufferAccumulator;
+
+    private final Timer timer = new Timer();
+
+    public OpenSearchSourceService(final OpenSearchSourceConfiguration sourceConfig,
+                                   final HostsService hostsService,
+                                   final OpenSearchService openSearchService,
+                                   final ElasticSearchService elasticSearchService,
+                                   final BufferAccumulator bufferAccumulator){
+        this.sourceConfig = sourceConfig;
+        this.hostsService = hostsService;
+        this.openSearchService = openSearchService;
+        this.elasticSearchService =elasticSearchService;
+        this.bufferAccumulator = bufferAccumulator;
+    }
+
+    public void processHosts(){
+        sourceConfig.getHosts().forEach(host ->{
```

Comment:
The hosts should be shared and provided to the OpenSearch Java client.

These `hosts` represent the different hosts which make up the OpenSearch cluster. They should not have their own schedules.